README.md

```diff
@@ -85,26 +151,97 @@ the Pre Deployment phase, which can be used by the scheduler to tell that a pod
 Workload Instances have a reference to the respective Deployment/StatefulSet/ReplicaSet, to check if it has reached the desired state. If it detects that the referenced object has reached
 its desired state (e.g. all pods of a deployment are up and running), it will be able to tell that a `PostDeploymentCheck` can be triggered.
 
-### Keptn Task
+### Keptn Task Definition
 
-tbd
+A `KeptnTaskDefinition` is a CRD used to define tasks that can be run by the Keptn Lifecycle Controller
+as part of pre- and post-deployment phases of a deployment.
+The task definition is a [Deno](https://deno.land/) script
+Please, refer to the [function runtime](./functions-runtime/) folder for more information about the runtime.
+In the future, we will support also other runtime, especially running a container image directly.
+
+A task definition can be configured in three different way:
+
+- inline
+- referring an HTTP script
+- referring another `KeptnTaskDefinition`
+
+An inline task definition looks like the following:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha1
+kind: KeptnTaskDefinition
+metadata:
+  name: deployment-hello
+spec:
+  function:
+    inline:
+      code: |
+        console.log("Deployment Task has been executed");
+```
 
-### Keptn Task Definition
+In the code section, it is possible to define a full-fletch Deno script.
+A further example, is available [here](./examples/taskonly-hello-keptn/inline/taskdefinition.yaml).
 
-tbd
+To runtime can also fetch the script on the fly from a remote webserver. For this, the CRD should look like the following:
```

Comment:
Needs explicit security considerations in the documentation. Can be done in a follow up pull request